nominal/core.py

```diff
@@ -311,6 +328,188 @@ def _from_conjure(cls, client: NominalClient, dataset: scout_catalog.EnrichedDat
         )
 
 
+@dataclass(frozen=True)
+class CreateChecklistVariable(BaseModel):
+    name: str
+    expression: str
+
+
+@dataclass(frozen=True)
+class CreateCheck(BaseModel):
+    name: str
+    priority: Priority
+    expression: str
+    description: str | None
+
+
+class ChecklistBuilder(BaseModel):
+    name: str
+    assignee_email: str
+    checklist_variables: Sequence[CreateChecklistVariable] = Field(default_factory=list)
+    checks: Sequence[CreateCheck] = Field(default_factory=list)
+    default_ref_name: str | None = None
+    commit_message: str | None = None
+    description: str | None = None
+    properties: Mapping[str, str] | None = None
+    labels: Sequence[str] = Field(default_factory=list)
+    _client: NominalClient = PrivateAttr()
```

Comment:
I had to change the signature quite a bit to make sure pydantic wasn't angry - namely with defaults for lists/dicts and the client not being serializable/deep-copy-able